### ISSUE TITLE
Add alf to homebrew

### ISSUE THREAD

**Issue by 0xdevalias** [DannyBen/alf#37]
Just came across this project and looks really neat! Would love it if it was added to homebrew so it was easier to install along with the rest of my tooling (without auto-running scripts from the internet)

You could also use one of the following to keep it up to date:

- https://github.com/marketplace/actions/bump-homebrew-formula (45 stars)
- https://github.com/dawidd6/action-homebrew-bump-formula (22 stars)

**Comment by DannyBen** [comment]
Thanks for the links. Alf is just a single file, so you don't even have to run the setup script from "the internet" you can just wget the alf file to your path.

Homebrew caters to mac users only, so it will just be the first packaging effort among others if we want to be thorough.

And speaking about different installation tools for different operating systems and packages, you might find [rush](https://github.com/DannyBen/rush-cli) useful. For example, I am installing alf by just running `rush alf`, and it simply runs [this recipe](https://github.com/DannyBen/rush-repo/blob/master/alf/main).

**Comment by 0xdevalias** [comment]
It's true that I could `wget` it, but I would be wanting to add this to my [dotfiles](https://github.com/0xdevalias/dotfiles) so I would be committing a fixed point in time and having to manually check for updates/etc, which I was hoping to avoid.

I will say, you have a bunch of really neat/useful looking projects! Rush looks really neat! Though then I would probably just shift the question to how do I bootstrap Rush in my repo and keep it up to date ;p

**Comment by DannyBen** [comment]
Yup. I agree that it would be nice to have one easy package manager to rule them all, but there isn't.

You prefer brew, on linuxes there are apt/yum, on ubuntu there is snap, and if you use a lot of open source tools from GitHub, many of them just give you a binary in the releases page.

So, although I do not object to having my tools wrapped in an easy to install package manager, it is a never ending effort which takes more time than developing the tools themselves sometimes.

This is why I developed rush by the way, and I install [all my tools](https://github.com/dannyben/rush-repo) in the same way: `rush <tool>`.

I did experiment with creating a brew formula some time ago, and found it to be too complex for the simple task of downloading a shell script.

Bottom line, I might try to create a brew package for my tools sometime, but I am not sure if and when.

**Comment by DannyBen** [comment]
[Offtopic]

By the way - I see you referenced from your dotfiles repo, and the need to avoid this pattern is another reason I made rush. 

Here is what I do:

1. Each of my rush packages, comes with its own configuration files if needed - for example, the [ruby package](https://github.com/DannyBen/rush-repo/tree/master/ruby) brings its `.gemrc` file.
2. Other personal configuration, is done in packages tucked away in the "my" subfolder - for example, the [package for configuring git](https://github.com/DannyBen/rush-repo/tree/master/my/gitconfig)
3. Since many tools I install require you to put something in your `~/.bashrc`, which seems like a terrible idea to me, I am only patching my `~/.bashrc` with one command that loads all files from `~/.bashrc.d/` directory, and then each package can just being its own configuration file. This is done in the [bashrc package](https://github.com/DannyBen/rush-repo/blob/master/bashrc/main) and an example use, can be seen in the [brew package](https://github.com/DannyBen/rush-repo/tree/master/brew), where it brings its own bashrc partial file.

Just thought I'd mention it, since it seems like you struggle with the same issues I have before I made these tools.

I can tell you that I am bootstrapping a new linux machine in minutes now, as opposed to hours and even days in the past (using my [bootstrap package](https://github.com/DannyBen/rush-repo/blob/master/bootstrap/main) as a starting point...)
